internal/util/url.go

```diff
@@ -80,27 +114,40 @@ func FetchResource(l *log.Logger, u url.URL) ([]byte, error) {
 
 		// check if present under oemDirPath, if so use it.
 		absPath := filepath.Join(oemDirPath, path)
-		data, err := ioutil.ReadFile(absPath)
+		f, err := os.Open(absPath)
+		var closer io.Closer = f
```

Comment:
oh, I see, you are using a single return. But if you just flip the conditionals you can put `if err == nil { return ... } else if !os.IsExistErr(err) { return err... }` at the top and remove one level of nesting from the bulk of the code.